### PR TITLE
Added crc to putfile

### PR DESCRIPTION
**Pull request by matwszedybyl**
added getting n setters in Putfile, applied setter in RPCRequestFactory and SDLProxyBase, CORRUPT_DATA to Result enum,  added tests.

### CODE REVIEW

File: sdl_android/src/main/java/com/smartdevicelink/proxy/SdlProxyBase.java [hunk 1]
```diff
@@ -1003,7 +1004,8 @@ private void sendOnSystemRequestToUrl(OnSystemRequest msg)
 		    	putFile.setCorrelationID(POLICIES_CORRELATION_ID);
 		    	putFile.setSdlFileName("response_data");
 		    	putFile.setFileData(response.toString().getBytes("UTF-8"));
-		    	updateBroadcastIntent(sendIntent, "DATA", "Data from cloud response: " + response.toString());
+				putFile.setCRC(response.toString().getBytes());
+				updateBroadcastIntent(sendIntent, "DATA", "Data from cloud response: " + response.toString());
```
Comment (brettywhite) [inline sdl_android/src/main/java/com/smartdevicelink/proxy/SdlProxyBase.java]: these dont need to be tabbed

File: sdl_android/src/main/java/com/smartdevicelink/proxy/rpc/PutFile.java [hunk 2]
```diff
@@ -116,8 +125,9 @@
     public static final String KEY_FILE_TYPE = "fileType";
     public static final String KEY_SDL_FILE_NAME = "syncFileName";
     public static final String KEY_OFFSET = "offset";
-    public static final String KEY_LENGTH = "length";
-    
+	public static final String KEY_LENGTH = "length";
+	public static final String KEY_CRC = "crc";
```
Comment (brettywhite) [inline sdl_android/src/main/java/com/smartdevicelink/proxy/rpc/PutFile.java]: no tabs

File: sdl_android/src/main/java/com/smartdevicelink/proxy/rpc/enums/Result.java [hunk 3]
```diff
@@ -168,7 +168,11 @@
     /**
      * The provided hash ID does not match the hash of the current set of registered data or the core could not resume the previous data.
      */
-    RESUME_FAILED;
+    RESUME_FAILED,
+	/**
+	 * The data sent failed to pass CRC check in receiver end
+	 */
+	CORRUPTED_DATA;
```
Comment (brettywhite) [inline sdl_android/src/main/java/com/smartdevicelink/proxy/rpc/enums/Result.java]: no tab

File: sdl_android/src/main/java/com/smartdevicelink/proxy/rpc/PutFile.java (hunk 2 above)
Comment (brettywhite) [inline sdl_android/src/main/java/com/smartdevicelink/proxy/rpc/PutFile.java]: no tabs

**Comment by brettywhite** [comment]
Do you need to also add the param for the other `buildPutFile` functions? [here](https://github.com/smartdevicelink/sdl_android/blob/25817e3db528bd2d8929a5fbc4276c0dec401c25/sdl_android/src/main/java/com/smartdevicelink/proxy/RPCRequestFactory.java#L493-L577)?

File: sdl_android/src/androidTest/java/com/smartdevicelink/test/rpc/requests/PutFileTests.java [hunk 4]
```diff
@@ -30,6 +30,7 @@ protected RPCMessage createMessage() {
 		msg.setSystemFile(Test.GENERAL_BOOLEAN);
 		msg.setOffset(Test.GENERAL_LONG);
 		msg.setLength(Test.GENERAL_LONG);
+		msg.setCRC(Test.GENERAL_LONG);
```
Comment (askirk) [inline sdl_android/src/androidTest/java/com/smartdevicelink/test/rpc/requests/PutFileTests.java]: Maybe another test where you use the setCRC(byte[] data) call?

**Comment by theresalech** [comment]
Resolves #452

**Comment by brettywhite** [comment]
@askirk @bilal-alsharifi @joeygrover 

This is ready for re-review

**Pull request merged**